tests/test_openclhpp.cpp

```diff
@@ -4517,4 +4517,41 @@ void testgetObjectInfo() {
     TEST_ASSERT_EQUAL(type, CL_GL_OBJECT_BUFFER);
     TEST_ASSERT_EQUAL(bufobj, 0);
 }
+static void *clEnqueueMapBuffer_testenqueueMapBuffer(
+    cl_command_queue command_queue, cl_mem buffer, cl_bool blocking_map,
+    cl_map_flags map_flags, size_t offset, size_t size,
+    cl_uint num_events_in_wait_list, const cl_event *event_wait_list,
+    cl_event *event, cl_int *errcode_ret, int num_calls) {
+    (void)command_queue;
+    TEST_ASSERT_EQUAL_PTR(make_mem(0), buffer);
+    TEST_ASSERT_EQUAL(CL_TRUE, blocking_map);
+    TEST_ASSERT_EQUAL(CL_MAP_WRITE, map_flags);
+    TEST_ASSERT_EQUAL(0, offset);
+    TEST_ASSERT_EQUAL(sizeof(int) * 1024, size);
+    TEST_ASSERT_EQUAL(0, num_events_in_wait_list);
+    TEST_ASSERT_EQUAL_PTR(nullptr, event_wait_list);
+    TEST_ASSERT_NOT_NULL(event);
```

Comment:
The event pointer is an optional output, so if the pointer is non-NULL then we should be writing a value to it (e.g. `make_event(N)`, and checking that the expected value was written in the calling function.

See, for example:

* https://github.com/KhronosGroup/OpenCL-CLHPP/blob/f96b8f865682e0bc27f6f61c34c5e75f76490ee4/tests/test_openclhpp.cpp#L3772-L3774
* https://github.com/KhronosGroup/OpenCL-CLHPP/blob/f96b8f865682e0bc27f6f61c34c5e75f76490ee4/tests/test_openclhpp.cpp#L3791